app/services/subscriptions/create_service.rb

```diff
@@ -42,21 +46,29 @@ def process_create
       return result.fail!('missing_argument', 'unable to find customer') unless current_customer
       return result.fail!('missing_argument', 'plan does not exists') unless current_plan
 
+      if currency_missmatch?(current_customer&.active_subscription&.plan, current_plan)
+        return result.fail!('currencies_does_not_match', 'currencies does not match')
+      end
+
+
       result.subscription = handle_subscription
       result
     rescue ActiveRecord::RecordInvalid => e
       result.fail_with_validations!(e.record)
     end
 
-    def current_subscription
-      @current_subscription ||= current_customer.subscriptions.active.first
+    def find_current_subscription(subscription_id)
+      return nil unless current_customer
+      return nil unless subscription_id
+
+      current_customer.subscriptions.active.find_by(id: subscription_id)
     end
 
     def handle_subscription
       return upgrade_subscription if upgrade?
       return downgrade_subscription if downgrade?
 
-      current_subscription || create_subscription
+      create_subscription
```

Comment:
I fear this change introduce a risk of duplication logic when called from the API.

If for some reason a request is sent twice to the endpoint, we will have two subscription instead of only one in the previous situation. Maybe we could have a constraint on the name or another system to handle idem-potency. What do you think?

Reply:
I see :thinking: 
Maybe we need a uniq attribute to handle this then. Like the `transaction_id` we have on event.

Reply:
added `unique_id`. I tried with `object_id` but since it's reserved word in rails I lost quite some time until I realised why nothing works properly :)

Reply:
I had validation so I thought that it would be enough but you are right, make sense to return existing 😅